prog/sensors/main.c

```diff
@@ -204,15 +219,25 @@ static int do_the_real_work(const sensors_chip_name *match, int *err)
 	int chip_nr;
 	int cnt = 0;
 
+	if (do_json)
+		printf("{\n");
 	chip_nr = 0;
 	while ((chip = sensors_get_detected_chips(match, &chip_nr))) {
 		if (do_sets) {
 			if (do_a_set(chip))
 				*err = 1;
 		} else
-			do_a_print(chip);
+			if (do_json) {
+				if (cnt > 0)
+					printf(",\n");
+				do_a_json_print(chip);
+			}
+			else
```

Comment:
Please use { } in both branches of the if statement for readability/consistency, and put "} else {" on a single line for consistency with the rest of the code.